### ISSUE TITLE
Most doclet properties on ES2015 constructors are ignored

### ISSUE THREAD

**Issue by 75lb** [jsdoc3/jsdoc#1129]
hi.. here's some output from a simple ES6 class..notice the second class record (the constructor) shows `undocumented: true` despite being documented.. 

```
  {
    "comment": "/**\n * a global class\n */",
    "meta": {
      "range": [
        26,
        385
      ],
      "filename": "es6-class.js",
      "lineno": 4,
      "path": "/Users/lloyd/Documents/75lb/testbed/input/global",
      "code": {
        "id": "astnode100000001",
        "name": "GlobalClass",
        "type": "ClassDeclaration",
        "paramnames": [
          "one",
          "two"
        ]
      }
    },
    "classdesc": "a global class",
    "name": "GlobalClass",
    "longname": "GlobalClass",
    "kind": "class",
    "scope": "global",
    "description": "the global class constructor",
    "params": [
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "first param",
        "name": "one"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "second param",
        "name": "two"
      }
    ]
  },
  {
    "comment": "/**\n   * the global class constructor\n   * @param {number} - first param\n   * @param {string} - second param\n   */",
    "meta": {
      "range": [
        165,
        246
      ],
      "filename": "es6-class.js",
      "lineno": 10,
      "path": "/Users/lloyd/Documents/75lb/testbed/input/global",
      "code": {
        "id": "astnode100000004",
        "name": "GlobalClass",
        "type": "MethodDefinition",
        "paramnames": [
          "one",
          "two"
        ]
      },
      "vars": {
        "": null
      }
    },
    "description": "the global class constructor",
    "params": [
      {
        "type": {
          "names": [
            "number"
          ]
        },
        "description": "first param",
        "name": "one"
      },
      {
        "type": {
          "names": [
            "string"
          ]
        },
        "description": "second param",
        "name": "two"
      }
    ],
    "name": "GlobalClass",
    "longname": "GlobalClass",
    "kind": "class",
    "scope": "global",
    "undocumented": true
  }
```

**Comment by hegemonic** [comment]
This is the intended behavior. For an ES2015 class, JSDoc creates two doclets: one for the class declaration, and another for the constructor. JSDoc then merges the second doclet into the first, so the second doclet is no longer needed. (We have to merge the doclets to prevent the class from being documented twice.)

**Comment by 75lb** [comment]
i had a look at the code, it seems only `description` and `params` are merged over - the exceptions are not.. (any `@throw` docs on the constructor are lost)

i'm still getting my head round the ES2015 differences.. for example, class instance members.. for *global* ES2015 classes, the instance members are a `memberof` the class declaration, e.g: 

```
    "longname": "GlobalClass#propOne",
    "kind": "member",
    "memberof": "GlobalClass",
```
but in a ES2015 class exported from a module, each instance member has a different parent - the constructor.. 
```
    "longname": "module:es6-class.ExportedClass#propOne",
    "kind": "member",
    "memberof": "module:es6-class.ExportedClass",
```

**Comment by hegemonic** [comment]
> i had a look at the code, it seems only `description` and `params` are merged over - the exceptions are not.. (any `@throw`, `@examples` etc docs on the constructor are lost)

You're right! I wasn't thinking about that when I implemented the feature. I think it's fair to say that this is a bug. I'm reopening this issue to address that bug.

> but in a ES2015 class exported from a module, each instance member has a different parent - the constructor..

Can you explain why you think this is wrong? To me, it seems like the exact same (correct) behavior you get with this ES5 code:

```js
/** @module es5-class */

/**
 * Exported class
 * @class
 */
exports.ExportedClass = function() {
    /** propOne */
    this.propOne = '';
};
```

**Comment by 75lb** [comment]
> Can you explain why you think this is wrong? To me, it seems like the exact same (correct) behavior you get with this ES5 code:

interesting, yes i now see that a directly exported ES5 class behaves the same way.. the direct `exports.something = function () {}` usage is fairly uncommon, which is possibly why i didn't spot this.. it's more commonly

```js
/** 
 * @module example 
 */
exports.something = something

/**
 * @alias module:example
 */
function something () {}
```

**Comment by hegemonic** [comment]
Fixed on master. The fix will be included in JSDoc 3.6.0.
